rotkehlchen/globaldb/handler.py

```diff
@@ -1550,77 +1554,75 @@ def hard_reset_assets_list(
+                    version = query.fetchone()
+                    if version is None or int(version[0]) != globaldb_get_setting_value(read_cursor, 'version', GLOBAL_DB_VERSION):  # noqa: E501
+                        msg = (
+                            'Failed to restore assets. Global database is not '
+                            'updated to the latest version'
+                        )
+                        return False, msg
+
+                with GlobalDBHandler().conn.write_ctx() as write_cursor:
+                    # If versions match drop tables
+                    write_cursor.execute('DELETE FROM assets')
+                    write_cursor.execute('DELETE FROM asset_collections')
+                    # Copy assets
+                    write_cursor.switch_foreign_keys('OFF')
+                    write_cursor.execute('INSERT INTO assets SELECT * FROM clean_db.assets;')
+                    write_cursor.execute('INSERT INTO evm_tokens SELECT * FROM clean_db.evm_tokens;')  # noqa: E501
+                    write_cursor.execute('INSERT INTO underlying_tokens_list SELECT * FROM clean_db.underlying_tokens_list;')  # noqa: E501
+                    write_cursor.execute('INSERT INTO common_asset_details SELECT * FROM clean_db.common_asset_details;')  # noqa: E501
+                    write_cursor.execute('INSERT INTO asset_collections SELECT * FROM clean_db.asset_collections')  # noqa: E501
+                    write_cursor.execute('INSERT INTO multiasset_mappings SELECT * FROM clean_db.multiasset_mappings')  # noqa: E501
+                    # Don't copy custom_assets since there are no custom assets in clean_db
+                    write_cursor.switch_foreign_keys('ON')
+
+                    with user_db.user_write() as user_db_cursor:
+                        user_db_cursor.switch_foreign_keys('OFF')
+                        user_db_cursor.execute('DELETE FROM assets;')
+                        # Get ids for assets to insert them in the user db
+                        write_cursor.execute('SELECT identifier from assets')
+                        ids = write_cursor.fetchall()
+                        ids_proccesed = ', '.join([f'("{id[0]}")' for id in ids])
+                        user_db_cursor.execute(f'INSERT INTO assets(identifier) VALUES {ids_proccesed};')  # noqa: E501
+                        user_db_cursor.switch_foreign_keys('ON')
+
+                with user_db.conn.read_ctx() as cursor:
+                    # Update the owned assets table
+                    user_db.update_owned_assets_in_globaldb(cursor)
+
+            except sqlite3.Error as e:
+                log.error(f'Failed to restore assets in globaldb due to {str(e)}')
+                return False, 'Failed to restore assets. Read logs to get more information.'
+            finally:  # Finally is executed even if there is a return in try or except blocks
```

Comment:
no reason to teach python to the reader

```suggestion
            finally:  # on the way out always detach the DB
```